### ISSUE TITLE
Incomplete error handling

### ISSUE THREAD

**Issue by peterbourgon** [richardartoul/molecule#7]
```go
func MessageEach(buffer *codec.Buffer, fn MessageEachFn) error {
	for !buffer.EOF() {
		fieldNum, wireType, err := buffer.DecodeTagAndWireType()
		if err == io.EOF {
			return nil
		}

		value, err := readValueFromBuffer(wireType, buffer)
```

Non-nil, non-io.EOF errors from the decode method are ignored. Oops! 😅

**Comment by richardartoul** [comment]
🤦 I will fix this soon. Thank you for catching!

**Comment by richardartoul** [comment]
Ok happily this should be resolved by #8 which just goes to show sometimes its best to procrastinate and let someone else solve the problem for you.

**Comment by peterbourgon** [comment]
As is tradition.
